network/src/peers/peers.rs

```diff
@@ -93,10 +93,11 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
             }
         }
 
-        // Attempt to connect to the default bootnodes of the network if the node has no active
+        // Attempt to connect to the default bootnodes and initial peers of the network if the node has no active
         // connections.
         if self.peer_book.get_active_peer_count() == 0 {
-            self.connect_to_bootnodes().await;
+            self.connect_to_peers(self.config.bootnodes()).await;
+            self.connect_to_peers(self.config.initial_peers()).await;
```

Comment:
since the initial peers shouldn't be used as bootnodes, but just as starting peers, they should only be connected to _only_ once, i.e. outside of `Node::update_peers` but somewhere else instead; if this wasn't a test-first functionality, the right place would be `main.rs`, but the test nodes should perhaps call this on their own once they're started?

Reply:
Agreed 👍